components/prism-stylus.js

```diff
@@ -0,0 +1,61 @@
+Prism.languages.stylus = {
+	'comment': {
+		pattern: /(^|[^\\])(\/\*[\w\W]*?\*\/|\/\/.*?(\r?\n|$))/g,
+		lookbehind: true
+	},
+	'keyword': /(px|r?em)\b|\b(is|defined|not|isnt|and|or|unless|for|in)\b/g,
+	'atrule': /@[\w-]+(?=\s+\S+)/gi,
+	'url': /url\((["']?).*?\1\)/gi,
+	'variable': /^\s*([\w-]+)(?=\s*[+-\\]?=)/gm,
+	'string': /("|')(\\\n|\\?.)*?\1/g,
+	'important': /\B!important\b/gi,
+	'hexcode': /#[\da-f]{3,6}/gi,
+	'entity': /\\[\da-f]{1,8}/gi,
+	'number': /\d+\.?\d*%?/g,
+	'selector': [
+		{
+			pattern: /::?after|before|first-letter|first-line|selection/g,
+			alias: 'pseudo-element'
+		},{
+			pattern: /:(?:active|checked|disabled|empty|enabled|first-child|first-of-type|focus|hover|in-range|invalid|lang|last-child|last-of-type|link|not|nth-child|nth-last-child|nth-last-of-type|nth-of-type|only-of-type|only-child|optional|out-of-range|read-only|read-write|required|root|target|valid|visited)(?:\(.*\))?/g,
+			alias:'pseudo-class'
+		},{
+			pattern: /(\[[\w-]+?\s*[*~|$^]?(?:=\s*\S+)?\])/g,
+			inside: {
+				"attr-name":
+				{
+					pattern: /(\[)([\w-]+)(?=\s*[~$^|=]{0,2})/g,
```

Comment:
Parentheses around `[\w-]+` are not needed.

Also, I think you forgot the `*=` operator. You should probably keep them always in the same order, for clarity.